### PR TITLE
fix(language selection): [MAG-3055] update language component with tick on active item

### PR DESCRIPTION
**Pull request by eTheMan80**
# <MAG-3055 / language selection>

## TL;DR

-   List
-   The
-   Core
-   Changes

## Details

Checklist / Reminders:

-   [x] Documentation / Story is created / adapted
-   [ ] Unit tests
-   [ ] CSS changes are BEM-compliant

Screenshot 2021-01-12 at 12 50 50

Screenshot 2021-01-12 at 17 25 48

### CODE REVIEW

**Comment by tricaalexandru** [comment]
@eTheMan80 should this still be in draft?

**Comment by eTheMan80** [comment]
@tricaalexandru yes was going to finalise but you asked me to jump on the bugs. I can finalise once I've finished my current card 🙂

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 1]
```diff
@@ -22,11 +24,22 @@ export const StyledLanguageMenuOption = styled.div`
...
+    setCurrentLabel(text)
+    onClick()
+  }
 
   return languages.map(language => (
-    <StyledLanguageMenuOption onClick={onClick} key={language.id}>
-      <CopyText tag="div" fontSize="sm" weight="regular" lineHeight="md" color="grey5" margin="0">
+    <StyledLanguageMenuOption onClick={() => handleItemClick(language.label)} key={language.id}>
+      {currentLabel === language.label && (
+        <StyledTickIcon width={14} height={10} color="grey5">
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:38]: @matthill8286 it seems I might be looking at a different file version to you

Screenshot 2021-01-12 at 10 44 11

File: src/components/Molecules/Dropdown/Dropdown.styled.ts [hunk 2]
```diff
@@ -77,6 +77,15 @@ export const StyledDone = styled(Icon)<IconPropsEnhanced>`
   left: 20px;
 `
 
+export const StyledTickIcon = styled(Icon)<IconPropsEnhanced>`
+  pointer-events: none;
+  position: absolute;
+  left: 15px;
+  display: flex;
```
[the remark this answers is not in the document]
Comment (eTheMan80) [reply src/components/Molecules/Dropdown/Dropdown.styled.ts]: @matthill8286 this has been corrected.

File: src/components/Molecules/LanguageMenu/LanguageMenu.tsx [hunk 3]
```diff
@@ -58,7 +60,9 @@ export const LanguageMenu: FC<LanguageMenuProps> = ({
     }
   })
 
-  const myLanguageContent = <LanguageMenuList languages={languages} onClick={handleLanguageClick} />
+  const myLanguageContent = (
+    <LanguageMenuList languages={languages} onClick={handleLanguageClick} languageId={languageId} />
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenu.tsx:64]: it's a index as opposed to an ID 👍 
```suggestion
    <LanguageMenuList languages={languages} onClick={handleLanguageClick} currentIndex={currentIndex} />
```

File: src/components/Molecules/LanguageMenu/LanguageMenu.tsx [hunk 4]
```diff
@@ -19,13 +19,15 @@ export interface LanguageMenuProps {
...
   extendedLanguageHandler: (item: LanguageMenuOption) => void
+  languageId?: number
 }
 
 export const LanguageMenu: FC<LanguageMenuProps> = ({
   className,
   languages = [],
   testIdSuffix,
   extendedLanguageHandler,
+  languageId,
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenu.tsx:30]:
```suggestion
  setupCurrentIndex = -1,
```

File: src/components/Molecules/LanguageMenu/LanguageMenu.tsx [hunk 5]
```diff
@@ -19,13 +19,15 @@ export interface LanguageMenuProps {
   languages: LanguageMenuOption[]
   testIdSuffix?: string
   extendedLanguageHandler: (item: LanguageMenuOption) => void
+  languageId?: number
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenu.tsx:22]:
```suggestion
  setupCurrentIndex?: number
```

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 6]
```diff
@@ -22,11 +24,22 @@ export const StyledLanguageMenuOption = styled.div`
 `
 
 export const LanguageMenuList = props => {
-  const { onClick, languages } = props
+  const { onClick, languages, languageId } = props
+  const [currentIndex, setCurrentIndex] = useState<number>(languageId)
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:28]: This needs to be outside the List.

File: src/components/Molecules/LanguageMenu/LanguageMenu.tsx (hunk 4 above)
Comment (matthill8286) [reply src/components/Molecules/LanguageMenu/LanguageMenu.tsx:30]: handleLanguageClick becomes :

``` 
const handleLanguageClick = (index, item) => {
    if (showDropdown) {
      toggleDropdown()
    }

    setCurrentIndex(index)

    if (extendedLanguageHandler) {
      extendedLanguageHandler(item)
    }
  }
```

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 7]
```diff
@@ -22,11 +24,22 @@ export const StyledLanguageMenuOption = styled.div`
 `
 
 export const LanguageMenuList = props => {
-  const { onClick, languages } = props
+  const { onClick, languages, languageId } = props
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:27]:
```suggestion
  const { onClick, languages, currentIndex } = props
```

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 8]
```diff
@@ -22,11 +24,22 @@ export const StyledLanguageMenuOption = styled.div`
...
-  return languages.map(language => (
-    <StyledLanguageMenuOption onClick={onClick} key={language.id}>
-      <CopyText tag="div" fontSize="sm" weight="regular" lineHeight="md" color="grey5" margin="0">
+  const handleItemClick = (index: number) => () => {
+    setCurrentIndex(index)
+    onClick(index)
+  }
+
+  return languages.map((language, index) => (
+    <StyledLanguageMenuOption onClick={handleItemClick(language.id)} key={language.id}>
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:36]:
```suggestion
    <StyledLanguageMenuOption onClick={() => onClick(index, language.id)} key={language.id}>
```

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 9]
```diff
@@ -22,11 +24,22 @@ export const StyledLanguageMenuOption = styled.div`
...
+  const { onClick, languages, languageId } = props
+  const [currentIndex, setCurrentIndex] = useState<number>(languageId)
 
-  return languages.map(language => (
-    <StyledLanguageMenuOption onClick={onClick} key={language.id}>
-      <CopyText tag="div" fontSize="sm" weight="regular" lineHeight="md" color="grey5" margin="0">
+  const handleItemClick = (index: number) => () => {
+    setCurrentIndex(index)
+    onClick(index)
+  }
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:33]:
```suggestion

```

File: src/components/Molecules/LanguageMenu/LanguageMenu.tsx (hunk 4 above)
Comment (matthill8286) [reply src/components/Molecules/LanguageMenu/LanguageMenu.tsx:30]: add:

```
const [currentIndex, setCurrentIndex] = useState<number>(setupCurrentIndex)
```

File: src/components/Molecules/LanguageMenu/LanguageMenuList.tsx [hunk 10]
```diff
@@ -1,13 +1,15 @@
-import React from 'react'
+import React, { useState } from 'react'
```
Comment (matthill8286) [inline src/components/Molecules/LanguageMenu/LanguageMenuList.tsx:1]:
```suggestion
import React from 'react'
```

File: src/components/Organisms/PageHeader/PageHeader.tsx [hunk 11]
```diff
@@ -22,7 +22,11 @@ export const PageHeader: React.FC<PageHeaderProps> = ({
         <Picture alt="thumbnail" onClick={onLogoClick} src={imageURL} width="120" height="60" />
 
         <StyledDropdownContainer>
-          <LanguageMenu extendedLanguageHandler={onLanguageChange} languages={languages} />
+          <LanguageMenu
+            extendedLanguageHandler={onLanguageChange}
+            languages={languages}
+            languageId={0}
```
Comment (matthill8286) [inline src/components/Organisms/PageHeader/PageHeader.tsx:28]:
```suggestion
            setupCurrentIndex={0}
```

File: src/components/Molecules/Dropdown/Dropdown.styled.ts [hunk 12]
```diff
@@ -77,6 +77,13 @@ export const StyledDone = styled(Icon)<IconPropsEnhanced>`
   left: 20px;
 `
 
+export const StyledTickIcon = styled(Icon)<IconPropsEnhanced>`
+  pointer-events: none;
+  position: absolute;
+  top: 14px;
+  left: 22px;
+`
+
```
Comment (matthill8286) [inline src/components/Molecules/Dropdown/Dropdown.styled.ts:86]: Add this to the styled language directory

**Review by tricaalexandru — APPROVED** [review]

**Comment by eTheMan80** [comment]
@matthill8286 can you review again??

**Review by matthill8286 — APPROVED** [review]

**Pull request merged**